crates/rpc/rpc-types-compat/src/transaction/mod.rs

```diff
@@ -1,117 +1,104 @@
-
-            (Some(gas_price), Some(signed_tx.max_fee_per_gas()))
+                GasPrice {
+                    gas_price: Some(gas_price),
+                    max_fee_per_gas: Some(signed_tx.max_fee_per_gas()),
+                }
+            }
+            _ => GasPrice::default(),
         }
-        _ => {
-            // OP-deposit
-            (Some(signed_tx.max_fee_per_gas()), None)
-        }
-    };
+    }
 
-    // let chain_id = signed_tx.chain_id().map(U64::from);
-    let chain_id = signed_tx.chain_id();
-    let blob_versioned_hashes = signed_tx.blob_versioned_hashes();
-    let access_list = signed_tx.access_list().cloned();
-    let authorization_list = signed_tx.authorization_list().map(|l| l.to_vec());
-
-    let signature =
-        from_primitive_signature(*signed_tx.signature(), signed_tx.tx_type(), signed_tx.chain_id());
-
-    WithOtherFields {
-        inner: Transaction {
-            hash: signed_tx.hash(),
-            nonce: signed_tx.nonce(),
-            from: signer,
-            to,
-            value: signed_tx.value(),
-            gas_price,
-            max_fee_per_gas,
-            max_priority_fee_per_gas: signed_tx.max_priority_fee_per_gas(),
-            signature: Some(signature),
-            gas: signed_tx.gas_limit() as u128,
-            input: signed_tx.input().clone(),
-            chain_id,
-            access_list,
-            transaction_type: Some(signed_tx.tx_type() as u8),
-            // These fields are set to None because they are not stored as part of the transaction
-            block_hash: tx_info.block_hash,
-            block_number: tx_info.block_number,
-            transaction_index: tx_info.index,
-            // EIP-4844 fields
-            max_fee_per_blob_gas: signed_tx.max_fee_per_blob_gas(),
-            blob_versioned_hashes,
-            // EIP-7702 fields
-            authorization_list,
-        },
-        // Optimism fields
-        #[cfg(feature = "optimism")]
-        other: reth_rpc_types::optimism::OptimismTransactionFields {
-            source_hash: signed_tx.source_hash(),
-            mint: signed_tx.mint(),
-            // only include is_system_tx if true: <https://github.com/ethereum-optimism/op-geth/blob/641e996a2dcf1f81bac9416cb6124f86a69f1de7/internal/ethapi/api.go#L1518-L1518>
-            is_system_tx: (signed_tx.is_deposit() && signed_tx.is_system_transaction())
-                .then_some(true),
-            deposit_receipt_version: None,
-        }
-        .into(),
-        #[cfg(not(feature = "optimism"))]
-        other: Default::default(),
+    /// Create a new rpc transaction result for a _pending_ signed transaction, setting block
+    /// environment related fields to `None`.
+    fn fill(tx: TransactionSignedEcRecovered, tx_inf: TransactionInfo) -> Self::Transaction;
+
+    /// Truncates the input of a transaction to only the first 4 bytes.
+    fn otterscan_api_truncate_input(tx: &mut Self::Transaction);
```

Comment:
I find this a bit weird, this could also be solved by introducing a trait for the the transaction and not baking this into the compat trait.

Reply:
yes absolutely, will open an issue, not to block anymore for alloy changes here. asides merge conflicts getting out of hand, I don't want to block this any longer since sdk work from arseni is moving fast and I'm uncertain if it's taking consideration of changes in this pr. I want to avoid anymore collisions by getting this in as is.